### PR TITLE
[DO NOT MERGE]: Fixes issues #8 to deploy merged pull requests

### PR DESCRIPTION
**Pull request by sgmeyer**
The extension relies on the `git.push` event.  When this event fires for a merged PR it is missing some of the necessary data in the API response.  Specifically, there isn't a way to get the commits associated with the merge.  Instead it uses the `git.pullrequest.merged` event and retrieves the commits by PR instead of Pull.  Doing so requires two hooks be setup in TFS and the code must handle the API responses differently.

### CODE REVIEW

**Comment by Verlic** [comment]
MD5: f1d3ff8443297732862df21dc4e57262

# NSP - Security Report

_This is an automated security audit of this project. Please do not modify its contents_

Vulnerabilities found: **4**
Affected Modules: **uglify-js, uglify-js, uglify-js, uglify-js**
Summary:

- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Incorrect Handling of Non-Boolean Comparisons During Minification
- Incorrect Handling of Non-Boolean Comparisons During Minification


## Detailed report
#### Regular Expression Denial of Service

Affected Module: **uglify-js**
Installed version: **2.2.5**
Patched versions: **>=2.6.0**
Advisory: https://nodesecurity.io/advisories/48
Path: **auth0-visualstudio-deploy-extension@2.2.3 > auth0-oauth2-express@1.3.0 > jade@1.11.0 > transformers@2.1.0 > uglify-js@2.2.5**

---

#### Regular Expression Denial of Service

Affected Module: **uglify-js**
Installed version: **2.2.5**
Patched versions: **>=2.6.0**
Advisory: https://nodesecurity.io/advisories/48
Path: **auth0-visualstudio-deploy-extension@2.2.3 > auth0-source-control-extension-tools@2.5.0 > auth0-oauth2-express@1.3.0 > jade@1.11.0 > transformers@2.1.0 > uglify-js@2.2.5**

---

#### Incorrect Handling of Non-Boolean Comparisons During Minification

Affected Module: **uglify-js**
Installed version: **2.2.5**
Patched versions: **>= 2.4.24**
Advisory: https://nodesecurity.io/advisories/39
Path: **auth0-visualstudio-deploy-extension@2.2.3 > auth0-oauth2-express@1.3.0 > jade@1.11.0 > transformers@2.1.0 > uglify-js@2.2.5**

---

#### Incorrect Handling of Non-Boolean Comparisons During Minification

Affected Module: **uglify-js**
Installed version: **2.2.5**
Patched versions: **>= 2.4.24**
Advisory: https://nodesecurity.io/advisories/39
Path: **auth0-visualstudio-deploy-extension@2.2.3 > auth0-source-control-extension-tools@2.5.0 > auth0-oauth2-express@1.3.0 > jade@1.11.0 > transformers@2.1.0 > uglify-js@2.2.5**

---

**Comment by sgmeyer** [comment]
Right now the Push event is being fired for both a push and a pr.  The extension needs to handle the scenario when a PR triggers a `git.push`.  Right now it reports a deployment failure.  Instead the event should either not be fired or the hook needs to ignore the event.  I have to further look at the API to see what our options are for this.

File: build/webpack/server.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ const logger = require('../../server/lib/logger');
 logger.info('Running development webpack server...');
 
 const options = {
+  disableHostCheck: true,
```
Comment (sgmeyer) [inline build/webpack/server.js:10]: This needs to be removed.

**Comment by sgmeyer** [comment]
Closing for an alternative solution

**Pull request closed without merging**